integration/cluster.go

```diff
@@ -0,0 +1,283 @@
+	resp, err := c.api.ListNodes(context.Background(), &api.ListNodesRequest{
+		Filters: &api.ListNodesRequest_Filters{
+			Roles: []api.NodeRole{api.NodeRoleManager},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, n := range resp.Nodes {
+		if n.ManagerStatus.Leader {
+			tn, ok := c.nodes[n.ID]
+			if !ok {
+				return nil, fmt.Errorf("leader id is %s, but it isn't found in test cluster object", n.ID)
+			}
+			return tn, nil
+		}
+	}
+	return nil, fmt.Errorf("cluster leader is not found in storage")
+}
+
+// RemoveNode removes node entirely. It tries to demote managers.
+func (c *Cluster) RemoveNode(id string) error {
+	node, ok := c.nodes[id]
+	if !ok {
+		return fmt.Errorf("remove node: node %s not found", id)
+	}
+	// demote before removal
+	if node.config.IsManager {
+		if err := c.SetNodeRole(id, api.NodeRoleWorker); err != nil {
+			return fmt.Errorf("demote manager: %v", err)
+		}
+
+	}
+	if err := node.Stop(); err != nil {
+		return err
+	}
+	delete(c.nodes, id)
+	if err := raftutils.PollFuncWithTimeout(nil, func() error {
+		resp, err := c.api.GetNode(context.Background(), &api.GetNodeRequest{NodeID: id})
+		if err != nil {
+			return fmt.Errorf("get node: %v", err)
+		}
+		if resp.Node.Status.State != api.NodeStatus_DOWN {
+			return fmt.Errorf("node %s is still not down", id)
+		}
+		return nil
+	}, opsTimeout); err != nil {
+		return err
+	}
+	if _, err := c.api.RemoveNode(context.Background(), &api.RemoveNodeRequest{NodeID: id}); err != nil {
+		return fmt.Errorf("remove node: %v", err)
+	}
+	return nil
+}
+
+// SetNodeRole sets role for node through control api.
+func (c *Cluster) SetNodeRole(id string, role api.NodeRole) error {
+	node, ok := c.nodes[id]
+	if !ok {
+		return fmt.Errorf("set node role: node %s not found", id)
+	}
+	if node.config.IsManager && role == api.NodeRoleManager {
+		return fmt.Errorf("node is already manager")
+	}
+	if !node.config.IsManager && role == api.NodeRoleWorker {
+		return fmt.Errorf("node is already worker")
+	}
+
+	var initialTimeout time.Duration
+	// version might change between get and update, so retry
+	for i := 0; i < 5; i++ {
+		time.Sleep(initialTimeout)
+		initialTimeout += 500 * time.Millisecond
+		resp, err := c.api.GetNode(context.Background(), &api.GetNodeRequest{NodeID: id})
+		if err != nil {
+			return err
+		}
+		spec := resp.Node.Spec.Copy()
+		spec.Role = role
+		if _, err := c.api.UpdateNode(context.Background(), &api.UpdateNodeRequest{
+			NodeID:      id,
+			Spec:        spec,
+			NodeVersion: &resp.Node.Meta.Version,
+		}); err != nil {
+			// there possible problems on calling update node because redirecting
+			// node or leader might want to shut down
+			if grpc.ErrorDesc(err) == "update out of sequence" || strings.Contains(grpc.ErrorDesc(err), "is closing") {
+				continue
+			}
+			// also there is possible another demote/promote is still in progress
+			if grpc.ErrorDesc(err) == "demote/promote in progress" {
```

Comment:
@LK4D4 Was this fixed by the recent cert and role changes or should we add a `FIXME`?